pkg/sensors/exec/exit_test.go

```diff
@@ -53,73 +52,75 @@ func TestExit(t *testing.T) {
 	assert.NoError(t, err)
 }
 
-func TestExitLeader(t *testing.T) {
-	var doneWG, readyWG sync.WaitGroup
-	defer doneWG.Wait()
-
-	ctx, cancel := context.WithTimeout(context.Background(), tus.Conf().CmdWaitTime)
-	defer cancel()
-
-	obs, err := observer.GetDefaultObserver(t, ctx, tus.Conf().TetragonLib)
-	if err != nil {
-		t.Fatalf("Failed to run observer: %s", err)
-	}
-	observer.LoopEvents(ctx, t, &doneWG, &readyWG, obs)
-	readyWG.Wait()
-
-	testExitLeader := testutils.RepoRootPath("contrib/tester-progs/exit-leader")
-
-	var startTime, exitTime time.Time
-
-	// The test executes 'exit-leader' benary which spawns a thread and
-	// exits the leader immediately while the new thread continues to
-	// run for 3 seconds and exits. We verify that we get exit event 3
-	// seconds after the start.
-
-	nextCheck := func(event ec.Event, l *logrus.Logger) (bool, error) {
-		switch ev := event.(type) {
-		case *tetragon.ProcessExec:
-			if ev.Process.Binary == testExitLeader {
-				startTime = ev.Process.StartTime.AsTime()
-			}
-			return false, nil
-		case *tetragon.ProcessExit:
-			if ev.Process.Binary == testExitLeader {
-				exitTime = ev.Time.AsTime()
-			}
-			return false, nil
-		}
-		return false, nil
-	}
-
-	finalCheck := func(l *logrus.Logger) error {
-		delta := exitTime.Sub(startTime)
-
-		fmt.Printf("execTime %v\n", startTime)
-		fmt.Printf("exitTime %v\n", exitTime)
-		fmt.Printf("delta %v\n", delta)
-
-		if delta < 3*time.Second {
-			return fmt.Errorf("unexpected delta < 3 seconds")
-		}
-		return nil
-	}
-
-	checker := &ec.FnEventChecker{
-		NextCheckFn:  nextCheck,
-		FinalCheckFn: finalCheck,
-	}
-
-	if err := exec.Command(testExitLeader).Run(); err != nil {
-		t.Fatalf("Failed to execute test binary: %s\n", err)
-	}
-
-	if err := jsonchecker.JsonTestCheck(t, checker); err != nil {
-		t.Logf("error: %s", err)
-		t.Fail()
-	}
-}
-
+// FIXME: Disable that for now as we revert https://github.com/cilium/tetragon/commit/1a37adf4bd400733292900335bf246f4263622bf
+// func TestExitLeader(t *testing.T) {
+// 	var doneWG, readyWG sync.WaitGroup
+// 	defer doneWG.Wait()
```

Comment:
Just as a note, I think t.Skip() would have been a better option here.

Reply:
I have created https://github.com/cilium/tetragon/pull/990 to fix that.